vignettes/datatable-benchmarking.Rmd

```diff
@@ -30,42 +24,104 @@ sudo lshw -class disk
 sudo hdparm -t /dev/sda
 ```
 
-## subset: index optimization switch off
+When comparing `fread` to non-R solutions be aware that R requires values of character columns to be added to _R's global string cache_, which adds significant overhead because R API for this is not thread safe, thus cannot be done in parallel.  
+
+# subset: threshold for index optimization on compound queries
 
-Index optimization will currently be turned off when doing subset using index and when cross product of elements provided to filter on exceeds > 1e4.
+Index optimization for compound filter queries will be not be used when cross product of elements provided to filter on exceeds 1e4 elements.  
+
+```r
+DT = data.table(V1=1:10, V2=1:10, V3=1:10, V4=1:10)
+setindex(DT)
+v = c(1L, rep(11L, 9))
+length(v)^4               # cross product of elements in filter
+#[1] 10000                # <= 10000
+DT[V1 %in% v & V2 %in% v & V3 %in% v & V4 %in% v, verbose=TRUE]
+#Optimized subsetting with index 'V1__V2__V3__V4'
+#on= matches existing index, using index
+#Starting bmerge ...done in 0.000sec 
+#...
+v = c(1L, rep(11L, 10))
+length(v)^4               # cross product of elements in filter
+#[1] 14641                # > 10000
+DT[V1 %in% v & V2 %in% v & V3 %in% v & V4 %in% v, verbose=TRUE]
+#Subsetting optimization disabled because the cross-product of RHS values exceeds 1e4, causing memory problems.
+#...
+```
 
-## subset: index aware benchmarking
+# subset: index aware benchmarking
 
-For convinience data.table automatically builds index on fields you are doing subset data. It will add some overhead to first subset on particular fields but greatly reduce time to query those columns in subsequent runs. When measuring speed best way is to measure index creation and query using index separately. Having such timings it is easy to decide what is the optimal strategy for your use case.
-To control usage of index use following options (see `?datatable.optimize` for more details):
+For convinience `data.table` automatically builds index on fields you are doing subset data. It will add some overhead to first subset on particular fields but greatly reduce time to query those columns in subsequent runs. When measuring speed best way is to measure index creation and query using index separately. Having such timings it is easy to decide what is the optimal strategy for your use case.  
```

Comment:
convenience

Reply:
Push to branch please :)

Reply:
Some other possible small grammar improvements:

~~builds index~~ => builds an index
~~greatly reduce time~~ => greatly reduces time
~~speed best way~~ => speed, the best way
~~using index separately~~ => using an index separately